src/python/pants/option/global_options.py

```diff
@@ -498,49 +507,46 @@ def register_options(cls, register):
     # global-scope options, for convenience.
     cls.register_bootstrap_options(register)
 
-    register('-x', '--time', type=bool,
-             help='Output a timing report at the end of the run.')
-    register('-e', '--explain', type=bool,
-             help='Explain the execution of goals.')
+    if register.bootstrap.v1:
+      register('-x', '--time', type=bool,
+               help='Output a timing report at the end of the run.')
+      register('-e', '--explain', type=bool,
+               help='Explain the execution of goals.')
     register('--tag', type=list, metavar='[+-]tag1,tag2,...',
              help="Include only targets with these tags (optional '+' prefix) or without these "
                   "tags ('-' prefix).  Useful with ::, to find subsets of targets "
                   "(e.g., integration tests.)")
 
-    # Toggles v1/v2 `Task` vs `@rule` pipelines on/off.
-    register('--v1', advanced=True, type=bool, default=True,
-             help='Enables execution of v1 Tasks.')
-    register('--v2', advanced=True, type=bool, default=False,
-             help='Enables execution of v2 @goal_rules.')
     register('--v2-ui', default=False, type=bool, daemon=False,
```

Comment:
Should this only be registered when `--v2` is True? If we make that change, then it'd be safe to change the default to True!